src/WebJobs.Script/Workers/Profiles/EnvironmentCondition.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.WebJobs.Script.Workers.Profiles;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers
+{
+    public class EnvironmentCondition : IWorkerProfileCondition
+    {
+        private readonly ILogger _logger;
+        private readonly IEnvironment _environment;
+        private readonly string _name;
+        private readonly string _expression;
+        private Regex _regex;
+
+        internal EnvironmentCondition(ILogger logger, IEnvironment environment, WorkerProfileConditionDescriptor descriptor)
+        {
+            if (descriptor is null)
+            {
+                throw new ArgumentNullException(nameof(descriptor));
+            }
+
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+
+            descriptor.Properties.TryGetValue(WorkerConstants.WorkerDescriptionProfileConditionName, out _name);
+            descriptor.Properties.TryGetValue(WorkerConstants.WorkerDescriptionProfileConditionExpression, out _expression);
+
+            Validate();
+        }
+
+        public string Name => _name;
+
+        public string Expression => _expression;
+
+        public bool Evaluate()
+        {
+            string value = _environment.GetEnvironmentVariable(Name);
+            if (string.IsNullOrEmpty(value))
```

Comment:
nit: new line between variable assignment and if condition

```suggestion

            if (string.IsNullOrEmpty(value))
```